gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -92,7 +92,7 @@ public MainViewModel(
     disposeOnClear(
         authenticationManager
             .getSignInState()
-            .compose(switchMapIfPresent(SignInState::getUser, userRepository::saveUser))
+            .compose(switchMapIfPresent(SignInState::value, userRepository::saveUser))
```

Comment:
From a reader's perspective, I feel that there's some information loss here.

Before, I could read the call on SignInState and immediately know it contains a `User` -- now this information is lost since we're just calling the generic method from the superclass.

Maybe we can override the `getter` for `value` in `SignInState` to keep the name `getUser` or at least provide a wrapper with this name to help readability?